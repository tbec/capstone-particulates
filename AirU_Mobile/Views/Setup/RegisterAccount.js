import React, {Component} from 'react';
import {View, Text, AsyncStorage,
         TextInput, KeyboardAvoidingView, Button, Image} from 'react-native';
import styles from '../../StyleSheets/Styles'
import { LOGIN_NAME, PASSWORD, TEST_MODE, WEB_URL } from '../../Components/Constants'
import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scrollview'

export default class RegisterAccount extends Component<Props> {
    constructor(props) {
        super(props)
        this.register = this.register.bind(this)
        this.webCall = this.webCall.bind(this)
        this.state = ({firstName: '', lastName: '', username: '', password: '', email: '', error: ''})
    }

    async register() {
        // make API call to login passing in login/password
        if (TEST_MODE && this.state.username != "TEST") {
            this.setState({ error: "Invalid username or password" })
            return
        }

        // check information
        let result = await this.webCall();
        res = JSON.parse(result);

        // check results
        if (res.success) {
            AsyncStorage.setItem(LOGIN_NAME, this.state.username);
            AsyncStorage.setItem(PASSWORD, this.state.password)
            this.props.navigation.navigate('ReviewFirst');
        } else {
            this.setState({error: 'Could not create account'})
        }
    }

    async webCall() {
        let urlBase = WEB_URL + '/register/user?'
        let user = 'username=' + this.state.username
        let firstName = '&firstname=' + this.state.firstName
        let lastName = '&lastname=' + this.state.lastName
        let email = '&email=' + this.state.email
        let password = '&password=' + this.state.password

        let url = urlBase + user + firstName + lastName + email + password

        console.log('URL: ' + url)
        let res = await fetch(url, {method: 'POST', credentials: 'include'})
        return res.json()
    }

    render() {
        return (
            <KeyboardAwareScrollView style={{backgroundColor: '#b3e6ff'}}>
                    <View style={{height: 200, alignItems: 'center', justifyContent: 'center', paddingTop: 20}}>
                        <Image source={require('../../Resources/red_cloud2.png')} 
                                        style={{width: '50%', height: '70%'}}/>
                    </View>
                    <KeyboardAvoidingView style={{flex: 1, flexDirection: 'column', justifyContent: 'center', alignItems: 'center'}}>
                    <Text style={styles.textInputLabel}>First Name</Text>
                        <TextInput editable={true} keyboardType='default' 
                                autoCorrect={false} placeholder='First Name' secureTextEntry={false} 
                                style={styles.textInput}
                                onChangeText={(value) => {this.setState({firstName: value})}}
                                />
                    </KeyboardAvoidingView>
                    <KeyboardAvoidingView style={{flex: 1, flexDirection: 'column', justifyContent: 'center', alignItems: 'center'}}>
                    <Text style={styles.textInputLabel}>Last Name</Text>
                        <TextInput editable={true} keyboardType='default' 
                                autoCorrect={false} placeholder='Last Name' secureTextEntry={false} 
                                style={styles.textInput}
                                onChangeText={(value) => {this.setState({lastName: value})}}
                                />
                    </KeyboardAvoidingView>
                    <KeyboardAvoidingView style={{flex: 1, flexDirection: 'column', justifyContent: 'center', alignItems: 'center'}}>
                    <Text style={styles.textInputLabel}>Username</Text>
                        <TextInput editable={true} keyboardType='default' 
                                autoCorrect={false} placeholder='Username' secureTextEntry={false} 
                                style={styles.textInput}
                                onChangeText={(value) => {this.setState({username: value})}}
                                />
                    </KeyboardAvoidingView>
                    <KeyboardAvoidingView style={{flex: 1, flexDirection: 'column', justifyContent: 'center', alignItems: 'center'}}>
                        <Text style={styles.textInputLabel}>Email</Text>
                        <TextInput editable={true} keyboardType='default' 
                                autoCorrect={false} placeholder='Email' secureTextEntry={false} 
                                style={styles.textInput}
                                onChangeText={(value) => {this.setState({email: value})}}
                                />
                    </KeyboardAvoidingView>
                    <KeyboardAvoidingView style={{flex: 1, flexDirection: 'column', justifyContent: 'center', alignItems: 'center'}}>
                        <Text style={styles.textInputLabel}>Password</Text>
                        <TextInput editable={true} keyboardType='default' 
                                autoCorrect={false} placeholder='Password' secureTextEntry={true} 
                                style={styles.textInput}
                                onChangeText={(value) => {this.setState({password: value})}}
                                />
                    </KeyboardAvoidingView>
                    <Text/>
                    <Text/>
                    <Button title="Register"
                        onPress={() => this.register()}
                        color='red' 
                        disabled={(this.state.login == "" || this.state.password == "" || this.state.email == "")}
                    />
                    <Text/>
                    <Text style={{height: 50, color: "red", textAlign: 'center'}}>
                        {this.state.error}
                    </Text>
            </KeyboardAwareScrollView>
        );
    }
}