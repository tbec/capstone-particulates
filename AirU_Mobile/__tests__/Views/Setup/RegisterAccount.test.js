import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import RegisterAccount from '../../../Views/Setup/RegisterAccount'

// used to find testIDs in code
let findById = function(tree, testID) {
    if(tree.props && tree.props.testID === testID) {
        return tree
    }
    if(tree.children && tree.children.length > 0)
    {
        let childs = tree.children
        for(let i = 0; i < childs.length; i++)
        {
            let item = findById(childs[i], testID)
            if(typeof(item) !== 'undefined') {
                return item
            }
        }
    }
}

test('Snapshot test for Confirmation Setup', () => {
    const snap = renderer.create(<RegisterAccount/>).toJSON();
    expect(snap).toMatchSnapshot();
})