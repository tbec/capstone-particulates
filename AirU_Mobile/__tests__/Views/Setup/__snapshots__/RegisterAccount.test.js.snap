// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for Confirmation Setup 1`] = `
<View>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../Resources/login_background.jpg",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <RCTScrollView
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      getTextInputRefs={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onScroll={[Function]}
      scrollEventThrottle={200}
      scrollToBottomOnKBShow={false}
      scrollToInputAdditionalOffset={75}
      startScrolledToBottom={false}
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 200,
              "justifyContent": "center",
              "paddingTop": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../Resources/aqLogo.png",
              }
            }
            style={
              Object {
                "height": "70%",
                "width": "50%",
              }
            }
          />
        </View>
        <View
          onLayout={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "red",
                "fontSize": 15,
                "fontWeight": "bold",
                "paddingBottom": 5,
                "paddingTop": 5,
              }
            }
          >
            First Name
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            editable={true}
            keyboardType="default"
            onChangeText={[Function]}
            placeholder="First Name"
            secureTextEntry={false}
            style={
              Object {
                "alignContent": "center",
                "backgroundColor": "white",
                "borderColor": "black",
                "borderWidth": 1,
                "height": 40,
                "justifyContent": "center",
                "paddingBottom": 10,
                "width": "50%",
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          onLayout={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "red",
                "fontSize": 15,
                "fontWeight": "bold",
                "paddingBottom": 5,
                "paddingTop": 5,
              }
            }
          >
            Last Name
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            editable={true}
            keyboardType="default"
            onChangeText={[Function]}
            placeholder="Last Name"
            secureTextEntry={false}
            style={
              Object {
                "alignContent": "center",
                "backgroundColor": "white",
                "borderColor": "black",
                "borderWidth": 1,
                "height": 40,
                "justifyContent": "center",
                "paddingBottom": 10,
                "width": "50%",
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          onLayout={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "red",
                "fontSize": 15,
                "fontWeight": "bold",
                "paddingBottom": 5,
                "paddingTop": 5,
              }
            }
          >
            Username
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            editable={true}
            keyboardType="default"
            onChangeText={[Function]}
            placeholder="Username"
            secureTextEntry={false}
            style={
              Object {
                "alignContent": "center",
                "backgroundColor": "white",
                "borderColor": "black",
                "borderWidth": 1,
                "height": 40,
                "justifyContent": "center",
                "paddingBottom": 10,
                "width": "50%",
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          onLayout={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "red",
                "fontSize": 15,
                "fontWeight": "bold",
                "paddingBottom": 5,
                "paddingTop": 5,
              }
            }
          >
            Email
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            editable={true}
            keyboardType="default"
            onChangeText={[Function]}
            placeholder="Email"
            secureTextEntry={false}
            style={
              Object {
                "alignContent": "center",
                "backgroundColor": "white",
                "borderColor": "black",
                "borderWidth": 1,
                "height": 40,
                "justifyContent": "center",
                "paddingBottom": 10,
                "width": "50%",
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          onLayout={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "red",
                "fontSize": 15,
                "fontWeight": "bold",
                "paddingBottom": 5,
                "paddingTop": 5,
              }
            }
          >
            Password
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            editable={true}
            keyboardType="default"
            onChangeText={[Function]}
            placeholder="Password"
            secureTextEntry={true}
            style={
              Object {
                "alignContent": "center",
                "backgroundColor": "white",
                "borderColor": "black",
                "borderWidth": 1,
                "height": 40,
                "justifyContent": "center",
                "paddingBottom": 10,
                "width": "50%",
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <Text />
        <Text />
        <Text
          style={
            Object {
              "color": "red",
              "height": 40,
              "textAlign": "center",
            }
          }
        >
          
        </Text>
        <Text />
        <View
          accessibilityRole="button"
          accessibilityStates={
            Array [
              "disabled",
            ]
          }
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
                Object {},
              ]
            }
          >
            <Text
              disabled={true}
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "red",
                  },
                  Object {
                    "color": "#cdcdcd",
                  },
                ]
              }
            >
              Register
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
